test/rubygems/test_gem_specification.rb

```diff
@@ -2820,8 +2820,22 @@ def test_validate_license
     end
 
     assert_match <<-warning, @ui.error
-WARNING:  licenses is empty, but is recommended.  Use a license abbreviation from:
-http://opensource.org/licenses/alphabetical
+WARNING:  licenses is empty, but is recommended.  Use a license identifier from:
+http://spdx.org/licenses
+    warning
+  end
+
+  def test_validate_license_values
+    util_setup_validate
+
+    use_ui @ui do
+      @a1.licenses = ['BSD']
+      @a1.validate
+    end
+
+    assert_match <<-warning, @ui.error
+WARNING: license value 'BSD' is invalid.  Use a license identifier from:
```

Comment:
@bf4, I'm not sure I can think of a good general rule that clearly compels "invalid" or "not recognized". Other warnings use "invalid" uniformly. But to the extent the patch might warn an author for using a recently added SPDX identifier that hasn't yet been added to RubyGem's list, that ID is "valid" per the English-language spec but not recognized against the hard-coded list. 

Concerns about "validity" and slippage between RubyGems and the SPDX list get philosophical pretty quick. Given the remote chances of such a scenario and the trifling extent of the resulting harm, I'm indifferent. If someone more invested in the project would like me to change it, I'll be happy to do so.